SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -656,6 +660,53 @@ void ActorSystem::EntityRemoved(const Worker_EntityId EntityId)
 	}
 }
 
+void ActorSystem::RegisterStablePathEntity(const Worker_EntityId EntityId)
+{
+	const UnrealMetadata& Metadata = ActorDataStore[EntityId].Metadata;
+
+	if (Metadata.StablyNamedRef.IsSet())
+	{
+		StablePathToEntityIdMap.Add(Metadata.StablyNamedRef.GetValue(), EntityId);
+	}
+}
+
+void ActorSystem::UnregisterStablePathEntity(const Worker_EntityId EntityId)
+{
+	const UnrealMetadata& Metadata = ActorDataStore[EntityId].Metadata;
+
+	if (Metadata.StablyNamedRef.IsSet())
+	{
+		StablePathToEntityIdMap.Remove(Metadata.StablyNamedRef.GetValue());
+	}
+}
+
+void ActorSystem::RestoreStablyNamedActor(const FUnrealObjectRef& ObjectRef, AActor* Actor)
+{
+	// This is called for startup actors in sublevels when they're loaded.
+	// In case the corresponding entities are still in our view (which can happen
+	// if the level is unloaded and immediately reloaded), we need to resolve the
+	// entities and create the actor channels.
+
+	if (const Worker_EntityId* EntityId = StablePathToEntityIdMap.Find(ObjectRef))
+	{
+		if (!NetDriver->PackageMap->ResolveEntityActor(Actor, *EntityId))
```

Comment:
Is this just safety checking, or is there a reason it might happen/

Reply:
Just for safety, based on the fact we were checking for this in `ReceiveActor`.